utilities/backupable/backupable_db.cc

```diff
@@ -2064,10 +2060,29 @@ Status BackupEngineImpl::BackupMeta::AddFile(
       return Status::Corruption("In memory metadata insertion error");
     }
   } else {
+    // Note: to save I/O, this check will pass trivially on already backed
+    // up files.
     if (itr->second->checksum_hex != file_info->checksum_hex) {
-      return Status::Corruption(
-          "Checksum mismatch for existing backup file. Delete old backups and "
-          "try again.");
+      std::string msg = "Checksum mismatch for existing backup file: ";
+      msg.append(file_info->filename);
+      msg.append(" Expected checksum is " + itr->second->checksum_hex +
+                 " while computed checksum is " + file_info->checksum_hex);
+      msg.append(
+          " If this DB file checks as not corrupt, try deleting old"
+          " backups or backing up to a different backup directory.");
+      return Status::Corruption(msg);
+    }
+    // Compare sizes, because we scanned that off the filesystem on both
+    // ends. This is like a check in VerifyBackup
+    if (itr->second->size != file_info->size) {
```

Comment:
Should we check size first instead of checksum?

Reply:
Sure. That makes sense. If there's a size mismatch, that is more "underlying cause" than checksum mismatch. I can make that (safe) change to provide better error reporting. :)

Reply:
And actually, with this change, the checksum check should never fail. Minor changes here are coming.